local/src/core/http.cc

```diff
@@ -776,6 +835,42 @@ Session::read_and_parse_request(swoc::FixedBufferWriter &buffer)
   return zret;
 }
 
+swoc::Errata
+Session::read_and_parse_proxy_hdr()
+{
+  // reading the PROXY data if any. A signle recv() should be sufficient as mentioned in the RFC
+  swoc::Errata errata;
+  swoc::LocalBufferWriter<MAX_PP_HDR_SIZE> w;
+  // Peek at the data to make sure it's a PROXY header
+  errata.note(S_DIAG, "Peeking at the socketdata to check PROXY header");
+  auto zret = peek(w.aux_span());
+  if (!zret.is_ok()) {
+    errata.note(S_ERROR, "Failed to peek at the socket data");
+    errata.note(std::move(zret.errata()));
+    return errata;
+  }
+  errata.note(std::move(zret.errata()));
+  if (zret.result() <= 0) {
+    // peek gets no data
+    return errata;
+  }
+  w.commit(zret.result());
+  // try to parse the PROXY header
+  ProxyProtocolUtil ppUtil;
+  errata.note(S_DIAG, "Attempting to parse PROXY header", w.size());
+  auto &&[ppNumBytes, ppParseErrata] = ppUtil.parse_header(w.view());
+  errata.note(std::move(ppParseErrata));
+  if (ppNumBytes > 0) {
+    errata.note(S_DIAG, "Got {} of pp bytes. consuming it from socket.", ppNumBytes);
+    char unusedBuf[ppNumBytes];
+    recv(_fd, unusedBuf, ppNumBytes, 0); // Peek at the data
+    // print the proxy message
+    errata.note(S_INFO, "{}", ppUtil);
+  } else {
+    errata.note(S_DIAG, "No valid PROXY header found, passing through.");
```

Comment:
nit: I'd remove "valid" here. It's a small thing, but we don't want to maybe imply that there was an invalid one.

Reply:
removed!